leetduel-backend/src/main.py

```diff
@@ -97,10 +97,17 @@ async def join_party(sid: str, data: dict) -> None:
 
     if party_code in parties and parties[party_code]["status"] == "waiting":
         parties[party_code]["players"].append({"sid": sid, "username": username})
-        await asyncio.sleep(3)
-        await sio.emit("player_joined", {"username": username}, room=party_code)
-        await asyncio.sleep(3)
+        player_usernames = [d["username"] for d in parties[party_code]["players"]]
         await sio.enter_room(sid, party_code)
+        await sio.emit(
+            "player_joined",
+            {
+                "username": username,
+                "players": player_usernames,
+            }, 
+            room=party_code,
+        )
+        print(f"players being added:\n{parties[party_code]["players"]}")
```

Comment:
switch order of `enter_room` and `player_joined`. otherwise, someone joining the room doesn't receive the `player_joined` socket event